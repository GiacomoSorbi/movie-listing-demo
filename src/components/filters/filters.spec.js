import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { genres } from '../../fixtures'

import Filters from './'

describe('Filters should', () => {
  it('exist', () => {
    expect(Filters).toBeDefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(
      <Filters
        genres={genres}
        currentMoviesGenres={[15, 18, 26]}
        onChangeGenreFilters={jest.fn()}
        onChangeMinRating={jest.fn()}
        value={'cat'}
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
