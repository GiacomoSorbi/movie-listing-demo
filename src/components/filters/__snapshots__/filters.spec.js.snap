// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should have a consistent structure 1`] = `
<Filters
  genres={
    Object {
      "12": "Feline Action",
      "15": "Feline Formation Novel",
      "18": "Feline Adventure",
      "19432": "Feline Philosophy",
      "23": "Feline Romance",
      "24": "Feline Western",
      "26": "Feline Drama",
    }
  }
  onChangeGenreFilters={[MockFunction]}
  onChangeMinRating={[MockFunction]}
  value="cat"
>
  <section
    className="movie-filters"
  >
    <div
      className="movie-filters-genres"
    >
      <Filter
        key="12"
        name="Feline Action"
        onChange={[MockFunction]}
        value="12"
      >
        <div
          className="movie-filter"
        >
          <input
            id="12"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="12"
          >
            Feline Action
          </label>
        </div>
      </Filter>
      <Filter
        key="15"
        name="Feline Formation Novel"
        onChange={[MockFunction]}
        value="15"
      >
        <div
          className="movie-filter"
        >
          <input
            id="15"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="15"
          >
            Feline Formation Novel
          </label>
        </div>
      </Filter>
      <Filter
        key="18"
        name="Feline Adventure"
        onChange={[MockFunction]}
        value="18"
      >
        <div
          className="movie-filter"
        >
          <input
            id="18"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="18"
          >
            Feline Adventure
          </label>
        </div>
      </Filter>
      <Filter
        key="23"
        name="Feline Romance"
        onChange={[MockFunction]}
        value="23"
      >
        <div
          className="movie-filter"
        >
          <input
            id="23"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="23"
          >
            Feline Romance
          </label>
        </div>
      </Filter>
      <Filter
        key="24"
        name="Feline Western"
        onChange={[MockFunction]}
        value="24"
      >
        <div
          className="movie-filter"
        >
          <input
            id="24"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="24"
          >
            Feline Western
          </label>
        </div>
      </Filter>
      <Filter
        key="26"
        name="Feline Drama"
        onChange={[MockFunction]}
        value="26"
      >
        <div
          className="movie-filter"
        >
          <input
            id="26"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="26"
          >
            Feline Drama
          </label>
        </div>
      </Filter>
      <Filter
        key="19432"
        name="Feline Philosophy"
        onChange={[MockFunction]}
        value="19432"
      >
        <div
          className="movie-filter"
        >
          <input
            id="19432"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <label
            htmlFor="19432"
          >
            Feline Philosophy
          </label>
        </div>
      </Filter>
    </div>
    <input
      defaultValue="3"
      id="min-rating"
      max="10"
      min="0"
      onChange={[MockFunction]}
      step="0.5"
      type="number"
    />
    <label
      htmlFor="min-rating"
    >
      Minimum Rating
    </label>
  </section>
</Filters>
`;
